File: src/run_avg_demo.py
import numpy as np
import movavgs as ma

num_samples = 50

# Truth process (constant):
truth = 220.6
truth_vec = truth*np.ones(num_samples)
# Estimation sequence and initial guess:
est = np.zeros(num_samples)

# Window size:
# Running average:  SMA window size=num_samples
w_size=num_samples
gain=1/w_size

# Generate noisy measurments
rg = np.random.default_rng(1)
mu=0
sigma = 0.5
meas = truth_vec + rg.normal(mu,sigma,num_samples)
est = ma.sma(meas,w_size)

if w_size<num_samples:
    avgtype = 'Simple Moving'
    title = avgtype + ' Average Estimator: Static Resistance (Window='+str(w_size)+')' 
else:
    avgtype = 'Running'
    title = avgtype + ' Average Estimator: Static Resistance' 
        

ylabel = 'Resistance (Ohms)'
ma.tme_plot(1,truth_vec, meas, est, ylabel, title, sigma)

File: src/matlab/runavgdemo.m
% running average demo
% Demo ideas:  1) Initial estimate under;  2) Initial estimate over;
% 3) large sigma;  4)  small sigma;  5) large sample size;  6) small sample
% size
mysigma = 0.5;
num_samples = 30;
truth = 220.6;
meas = truth + mysigma*randn(1,num_samples);
truth_vec = truth*ones(1,num_samples);
est = zeros(1,num_samples);
est(1)=210;
for k=2:num_samples
    prior_est = est(k-1);
    gain = 1/k;
    est(k) = recursive_avg(meas(k),gain,prior_est);
end
figure(1)
plot(1:num_samples,truth_vec,'b--*',1:num_samples,meas,'r--o',1:num_samples,est,'d:k');
legend('Truth','Measurements','Estimates','Location','southeast');
xlabel('Time')
ylabel('Resistance (Ohms)')
title('Simple Running Average Estimator: Resistor')
display('Error: ')
display(est(num_samples)-truth)

